app/models/course.rb

```diff
@@ -195,6 +195,48 @@ def sanitized_name
   def invalidate_cgdubs
     cgdub_dependencies_updated
     save!
+
+    update_course_gdu_watchlist_instances
+  end
+
+  # Update the grace day usage condition watchlist instances for each course user datum
+  # This is called when:
+  # - Grace days or late slack have been changed and the record is saved
+  # - invalidate_cgdubs are somehow incurred
+  def update_course_gdu_watchlist_instances
```

Comment:
I suggest that this function be moved into the watchlist instance model instead. This function modifies the state of the watchlist instance, not really that of the course